electrum/lnchannel.py

```diff
@@ -763,13 +750,22 @@ def is_static_remotekey_enabled(self) -> bool:
         channel_type = ChannelType(self.storage.get('channel_type'))
         return bool(channel_type & ChannelType.OPTION_STATIC_REMOTEKEY)
 
+    @property
+    def sweep_address(self) -> str:
+        # TODO: in case of unilateral close with pending HTLCs, this address will be reused
+        addr = None
+        assert self.is_static_remotekey_enabled()
```

Comment:
Re the `"remove support for channels without static remote pubkey"` commit in general, I think we might need to think about this a bit more.

The reason we want to do it, is to simplify the code, and the justification is that most likely hardly anyone has wallets with legacy channels in it (as these were never in a released version). However, what should the behaviour be if someone does have such a channel?
Should we maybe put an assert in `LNWallet.__init__` that all channels are `is_static_remotekey_enabled()`, so that at least there is clear and well-defined behaviour?
(although ofc then the wallet cannot be opened even just to delete those channels)

In particular, I do have a mainnet wallet that has a few legacy channels saved in it -- all of them are already redeemed (deeply-closed). When I open that wallet with this PR, there is no user-visible error in the GUI, but the logs are spammed with AssertionErrors, and in fact I expect lnwatcher to be broken even for the non-legacy `OPEN` channels...

<details>
 <summary>tracebacks from log</summary>

```
E/W | lnwatcher.LNWalletWatcher.[default_wallet-LNW] | Exception in do_breach_remedy: AssertionError()
Traceback (most recent call last):
  File "...\electrum\electrum\util.py", line 1235, in wrapper
    return await func(*args, **kwargs)
  File "...\electrum\electrum\lnwatcher.py", line 443, in do_breach_remedy
    sweep_info_dict = chan.sweep_ctx(closing_tx)
  File "...\electrum\electrum\lnchannel.py", line 264, in sweep_ctx
    our_sweep_info = self.create_sweeptxs_for_our_ctx(ctx)
  File "...\electrum\electrum\lnchannel.py", line 253, in create_sweeptxs_for_our_ctx
    return create_sweeptxs_for_our_ctx(chan=self, ctx=ctx, sweep_address=self.sweep_address)
  File "...\electrum\electrum\lnchannel.py", line 757, in sweep_address
    assert self.is_static_remotekey_enabled()
AssertionError
```
```
E/W | lnwatcher.LNWalletWatcher.[default_wallet-LNW] | Exception in trigger_callbacks: AssertionError()
Traceback (most recent call last):
  File "...\electrum\electrum\util.py", line 1235, in wrapper
    return await func(*args, **kwargs)
  File "...\electrum\electrum\lnwatcher.py", line 208, in trigger_callbacks
    await callback()
  File "...\electrum\electrum\lnwatcher.py", line 225, in check_onchain_situation
    keep_watching = await self.do_breach_remedy(funding_outpoint, closing_tx, spenders)
  File "...\electrum\electrum\util.py", line 1235, in wrapper
    return await func(*args, **kwargs)
  File "...\electrum\electrum\lnwatcher.py", line 443, in do_breach_remedy
    sweep_info_dict = chan.sweep_ctx(closing_tx)
  File "...\electrum\electrum\lnchannel.py", line 264, in sweep_ctx
    our_sweep_info = self.create_sweeptxs_for_our_ctx(ctx)
  File "...\electrum\electrum\lnchannel.py", line 253, in create_sweeptxs_for_our_ctx
    return create_sweeptxs_for_our_ctx(chan=self, ctx=ctx, sweep_address=self.sweep_address)
  File "...\electrum\electrum\lnchannel.py", line 757, in sweep_address
    assert self.is_static_remotekey_enabled()
AssertionError
```
</details>

Reply:
I've added a commit https://github.com/spesmilo/electrum/commit/2175bb6df09938a0d4f62058ad7fe823a1684ffb to the branch that contains a wallet DB upgrade that enforces there are no "legacy" type channels.